test/framework/base_unit_test_case.py

```diff
@@ -19,14 +19,13 @@ class BaseUnitTestCase(TestCase):
     _base_setup_called = False
     # This allows test classes (e.g., TestNetwork) to disable network-related patches for testing the patched code.
     _do_network_mocks = True
-    _repatchable_items = {}
     _fake_hostname = 'fake_hostname'
 
     def setUp(self):
         super().setUp()
         self.addCleanup(patch.stopall)
 
-        self._repatchable_items = {}  # Reset this at the start of each test.
+        self._patched_items = {}
```

Comment:
Functionality is now:
 - (unchanged) still only allow a patched item to be repatched if it was patched with `allow_repatch=True`
 - (changed) now anything that has been `patch()`ed can be `unpatch()`ed

This change buys us the ability to patch something for *every* test by patching in `BaseUnitTestCase` and then unpatch it for a specific test. This usually makes more sense than needing to patch the offending item in a bunch of unrelated test classes. (Basically this allows us to whitelist instead of blacklist -- items can be patched by default instead of unpatched by default.)